daemon/algod/api/server/v2/dryrun_test.go

```diff
@@ -1634,3 +1656,148 @@ int 1`)
 		logResponse(t, &response)
 	}
 }
+
+func checkStateDelta(t *testing.T,
+	response *generated.StateDelta,
+	expectedDelta *generated.StateDelta,
+) {
+	for i, vd := range *response {
+		assert.Equal(t, (*expectedDelta)[i].Key, vd.Key)
+
+		// Pointer checks: make sure we don't try to derefence a nil.
+		if (*expectedDelta)[i].Value.Bytes == nil {
+			assert.Nil(t, vd.Value.Bytes)
+		} else {
+			assert.NotNil(t, vd.Value.Bytes)
+			assert.Equal(t, *(*expectedDelta)[i].Value.Bytes, *vd.Value.Bytes)
```

Comment:
This seems to be appear consistently in this file -- so I don't necessarily have a problem with you following the same convention -- but I don't understand why we're using pointers everywhere. Afaict, most uses don't need to be pointers, especially with slice types like generated.StateDelta.

Reply:
Good point - simplified some tests there